tests/recorder.js

```diff
@@ -0,0 +1,113 @@
+*/
+
+var nock = require('nock');
+var fs = require('fs');
+var request = require('request');
+var yargs = require('yargs').argv;
+
+var testUtils = require('./test-utils');
+var logger = require('../logger');
+
+// Run the server
+require('../bin/www');
+// Process Redis tasks
+require('../worker');
+
+var timeout = 10000;  // 10 seconds
+var testNumber;
+
+// Require that a user provide the test number (which represents the fixtures/{testno})
+// via the command line.  If none provided, bail.
+if(yargs.test && !isNaN(yargs.test)) {
+    testNumber = yargs.test;
+}
+else {
+    logAndQuit('No test provided, exiting.', process.argv);
+}
+
+if(!yargs.description) {
+    logAndQuit('No description provided, exiting.', process.argv);
+}
+
+// Start recording the requests sent to GitHub as well as the responses sent back
+nock.recorder.rec({ output_objects: true });
+
+// Kick off!  Simulates us receiving a POST from GitHub!
+request.post({
+    url: testUtils.appHost + '/hook',
+    headers: testUtils.getFileContents(testNumber, 'headers'),
+    form: testUtils.getFileContents(testNumber, 'payload')
+}, function(error) {
+    if(error) logAndQuit('Post for kicking off the recording could not be completed: ', error);
+});
+
+// Since we go into this recording "blindly", the only way we know it's done is
+// by setting a timeout which cuts the testing off after a given point.
+// If the test doesn't resolve in this timeout, GitHub must be very slow or
+// the test is too complex, in which case it's best to simply change the timeout
+// value on your local machine
+setTimeout(function() {
+
+    var filepath = testUtils.fixturesDir + testNumber + '/';
+    var manifest = {
+        urls: {},
+        description: yargs.description || '',
+        posts: 0 // Will increment as we POST comments
+    };
+
+    // Calling play() provides an array of recorded URL calls with their payload
+    nock.recorder.play().forEach(function(obj) {
+
+        var filename = obj.path.replace(/\//g, '-').substr(1) + '.json';
+        var contents;
+
+        // Don't bother recording the initial request to the hook
+        if(obj.scope.indexOf(testUtils.appHost) !== -1) return;
+
+        // Add this to the registry
+        manifest.urls[obj.path] = {
+            file: filename,
+            method: obj.method
+        };
+
+        // Increment posts if need be
+        if(obj.method.toLowerCase() === 'post') manifest.posts++;
+
+        // Generate the file contents; not all are JSON (file contents)
+        // so we'll use a try/catch
+        try {
+            contents = toJson(obj.response);
+        }
+        catch(ex) {
+            contents = obj;
+        }
+        fs.writeFileSync(filepath + filename, contents);
+    });
+
+    // Write the manifest file
+    fs.writeFileSync(filepath + 'manifest.json', toJson(manifest));
+
+    process.exit();
+}, timeout);
+
+// Logs and quits
+function logAndQuit() {
+    logger.error(arguments);
+    process.exit();
```

Comment:
To indicate a failure, we could exit with `process.exit(1)`.

https://nodejs.org/api/process.html#process_process_exit_code